test/e2e/live_iso_test.go

```diff
@@ -0,0 +1,103 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	bmov1alpha1 "github.com/metal3-io/baremetal-operator/apis/metal3.io/v1alpha1"
+	infrav1 "github.com/metal3-io/cluster-api-provider-metal3/api/v1beta1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	clusterv1alpha4 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("When testing live iso [live-iso]", func() {
+	liveIsoTest()
+})
+
+func liveIsoTest() {
+	BeforeEach(func() {
+		validateGlobals(specName)
+		// We need to override clusterctl apply log folder to avoid getting our credentials exposed.
+		clusterctlLogFolder = filepath.Join(os.TempDir(), "clusters", bootstrapClusterProxy.GetName())
+	})
+
+	It("Should create a target cluster node with live iso", func() {
+		controlPlaneMachineCount = 1
+		workerMachineCount = 1
+
+		workloadClusterTemplate := clusterctl.ConfigCluster(ctx, clusterctl.ConfigClusterInput{
+			KubeconfigPath:           bootstrapClusterProxy.GetKubeconfigPath(),
+			ClusterctlConfigPath:     clusterctlConfigPath,
+			Flavor:                   "live-iso",
+			Namespace:                namespace,
+			ClusterName:              clusterName,
+			KubernetesVersion:        e2eConfig.GetVariable(KubernetesVersion),
+			ControlPlaneMachineCount: &controlPlaneMachineCount,
+			WorkerMachineCount:       &workerMachineCount,
+			InfrastructureProvider:   clusterctl.DefaultInfrastructureProvider,
+			LogFolder:                clusterctlLogFolder,
+		})
+
+		Expect(workloadClusterTemplate).ToNot(BeNil(), "Failed to get the cluster template")
+		Expect(bootstrapClusterProxy.Apply(ctx, workloadClusterTemplate)).To(Succeed())
+
+		By("Waiting for the control plane to be running and worker node to be in provisioning state")
+		Eventually(func() (int, error) {
+			n := 0
+			machineList := &clusterv1alpha4.MachineList{}
+			if err := bootstrapClusterProxy.GetClient().List(ctx, machineList, client.InNamespace(namespace), client.MatchingLabels{clusterv1.ClusterLabelName: clusterName}); err == nil {
+				for _, machine := range machineList.Items {
+					if isWorkerNode(machine) && strings.EqualFold(machine.Status.Phase, "provisioning") {
+						n++
+					} else if strings.EqualFold(machine.Status.Phase, "running") {
+						n++
+					}
+				}
+			}
+			return n, nil
+		}, e2eConfig.GetIntervals(specName, "wait-worker-nodes")...).Should(Equal(2))
```

Comment:
Since you are only verifying the kcp machine is running I believe we can simplify the condition and remove the fuction isWorker as follow:
```suggestion
		Eventually(func() (bool) {
			machineList := &clusterv1alpha4.MachineList{}
			if err := bootstrapClusterProxy.GetClient().List(ctx, machineList, client.InNamespace(namespace), client.MatchingLabels{clusterv1.ClusterLabelName: clusterName}); err == nil {
				for _, machine := range machineList.Items {
					if strings.EqualFold(machine.Status.Phase, "running") {
						return true
					}
				}
			}
			return false
		}, e2eConfig.GetIntervals(specName, "wait-worker-nodes")...).To(Succeed())
```

Reply:
I tried the test by removing the check for worker node, but it started failing when I tried to get worker node after this loop, so I think we have to keep it there. Let me know if it can be improved in some other way.

Reply:
can you please describe what was the issue ?